packages/api/src/gateways/edge-gateway/abstract_edge_gateway_strategy.ts

```diff
@@ -4,13 +4,16 @@ import { WalletWithBalance } from "@cryptify/common/src/domain/wallet_with_balan
 import { GetWalletsRequest } from "@cryptify/common/src/requests/get_wallet_request";
 import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
 import { GetTransactionsRequest } from "@cryptify/common/src/requests/get_transaction_request";
+import { DeleteWalletRequest } from "@cryptify/common/src/requests/delete_wallet_request";
+import { Wallet } from "@cryptify/common/src/domain/entities/wallet";
 
 export abstract class AbstractEdgeGatewayStrategy extends AbstractServiceGateway {
     protected constructor(uri: string) {
         super(uri);
     }
 
     abstract createWallet(req: CreateWalletRequest): Promise<WalletWithBalance>;
+    abstract deleteWallet(req: DeleteWalletRequest): Promise<Wallet>;
```

Comment:
Again if youre going to return the wallet it should be with the balance

```suggestion
    abstract deleteWallet(req: DeleteWalletRequest): Promise<WalletWithBalance>;
```